### PR TITLE
FORMS-14016 - Bug fixes of short description

### PR DESCRIPTION
**Pull request by TalmizAhmed**
Jira: https://jira.corp.adobe.com/browse/FORMS-14016

Screenshot 2024-04-23 at 11 33 58 AM
Screenshot 2024-04-23 at 11 34 07 AM

Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--aem-boilerplate-forms--adobe-rnd.hlx.page/drafts/talmiz/form
- After: https://short-desc--aem-boilerplate-forms--adobe-rnd.hlx.page/drafts/talmiz/form

### CODE REVIEW

**Comment by vdua** [comment]
The url should be of the form 
https://short-desc--aem-boilerplate-forms--adobe-rnd.hlx.page/content/forms/af/talmiz/eds

**Comment by vdua** [comment]
also the build is failing

**Comment by TalmizAhmed** [comment]
> The url should be of the form https://short-desc--aem-boilerplate-forms--adobe-rnd.hlx.page/content/forms/af/talmiz/eds

@vdua The fstab in boilerplate is configured to a gdrive but the bug is a crosswalk specific bug. There is no short description(tooltip) in doc based authoring according to this: https://main--afb--adobe.hlx.live/docs/docbased/formdef so I had to resort to a form from crosswalk

File: blocks/form/util.js [hunk 1]
```diff
@@ -11,8 +11,11 @@ export function stripTags(input, allowd = allowedTags) {
     .join(''); // making sure the allowed arg is a string containing only tags in lowercase (<a><b><c>)
   const tags = /<\/?([a-z][a-z0-9]*)\b[^>]*>/gi;
   const comments = /<!--[\s\S]*?-->/gi;
+  const nbsp = /&nbsp;/g; // nbsp: non-breaking space character
   return input.replace(comments, '')
-    .replace(tags, ($0, $1) => (allowed.indexOf(`<${$1.toLowerCase()}>`) > -1 ? $0 : ''));
+    .replace(tags, ($0, $1) => (allowed.indexOf(`<${$1.toLowerCase()}>`) > -1 ? $0 : ''))
+    .replace(nbsp, '')
+    .trim();
```
Comment (vdua) [inline blocks/form/util.js:21]: if input is not a string return the same back to prevent NPE

Comment (TalmizAhmed) [reply blocks/form/util.js:21]: done

File: blocks/form/util.js [hunk 2]
```diff
@@ -5,14 +5,20 @@ const headings = Array.from({ length: 5 }, (_, i) => `<h${i + 1}>`).join('');
 const allowedTags = `${headings}<a><b><p><i><em><strong><ul><li>`;
```
Comment (vdua) [inline blocks/form/util.js]: add `<ol>` in the allowed tags as well.

**Review by vdua — DISMISSED** [review]

File: blocks/form/util.js [hunk 3]
```diff
@@ -68,8 +74,8 @@ export function createLabel(fd, tagName = 'label') {
     if (fd.label.visible === false) {
       label.dataset.visible = 'false';
     }
-    if (fd.Tooltip) {
-      label.title = fd.Tooltip;
+    if (fd.tooltip) {
+      label.title = stripTags(fd.tooltip, '');
```
Comment (vdua) [inline blocks/form/util.js:78]: test case where tooltip has invalid tags/valid tags is missing

Comment (TalmizAhmed) [reply blocks/form/util.js:78]: @vdua Regardless of whether valid/invalid tags are give, all of them are stripped out due to the second param on line 78

Comment (vdua) [reply blocks/form/util.js:78]: I am saying that if you have a tooltip as `<div><b>something</b><div>` or `This is my title containing<script>xyz</xyz>` then what should be the title attribute ?That test case is missing

File: test/unit/fixtures/components/text-input/text-input-advance.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ export const fieldDef = {
     autoComplete: 'off',
     placeholder: 'Aya Tan',
     description: '<p>Hint - First name should be minimum 3 characters and a maximum of 10 characters.</p>',
-    tooltip: '<p>Enter your first name.</p>',
+    tooltip: '<script><p>Enter your first name.</p></script>',
```
Comment (TalmizAhmed) [inline test/unit/fixtures/components/text-input/text-input-advance.js:15]: @vdua Added extra tags

File: blocks/form/util.js (hunk 3 above)
Comment (TalmizAhmed) [reply blocks/form/util.js:78]: Already exists, but adding more.

**Review by jalagari — APPROVED** [review]

**Pull request merged**